src/librustc/session/mod.rs

```diff
@@ -437,6 +437,12 @@ impl Session {
     pub fn print_llvm_passes(&self) -> bool {
         self.opts.debugging_opts.print_llvm_passes
     }
+    pub fn nll(&self) -> bool {
+        self.features.borrow().nll || self.opts.debugging_opts.nll
+    }
+    pub fn nll_dump_cause(&self) -> bool {
+        self.features.borrow().nll || self.opts.debugging_opts.nll_dump_cause
```

Comment:
I am debating whether we should have `#![feature(nll)]` enable dump-cause or not. The main reason not to is that it is **so sloooooow** right now. I feel like we might want to enable it once we make it faster (and then we wouldn't have this switch at all.)